src/_guides/language/effective-dart/design.md

```diff
@@ -1005,90 +1089,254 @@ var names = people.map((Person person) {
+
+  FilteredObservable(this._predicate, this._observers);
+
+  [!void Function(Event)!] notify(Event event) {
+    if (!_predicate(event)) return null;
+
+    [!void Function(Event)!] last;
+    for (var observer in _observers) {
+      observer(event);
+      last = observer;
+    }
+
+    return last;
+  }
+}
+{% endprettify %}
+
+It may still be worth defining a typedef if the function type is particularly
+long or frequently used. But in most cases, users want to see what the function
+type actually is right where it's used, and the function type syntax gives them
+that clarity.
+
+
+### CONSIDER using the generalized function type syntax for parameters.
+
+Dart has a special syntax when defining a parameter whose type is a function.
+Sort of like C, you surround the parameter's name with the function's return
+type and parameter signature:
+
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (function-type-param)"?>
+{% prettify dart %}
+Iterable<T> where(bool predicate(T element)) => ...
+{% endprettify %}
+
+Before Dart 2 added general function type syntax, this was the only way to give
+a parameter a function type without defining a typedef. Now that Dart has a
+general notation for function types, you can use it for function-typed
+parameters as well:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (function-type-param)"?>
 {% prettify dart %}
-bool isValidString(String value, Function predicate) => ...
+Iterable<T> where(bool Function(T element) predicate) => ...
 {% endprettify %}
 
-One exception is if the variable can be one of several different function types.
-For example, it may allow a function that takes one mandatory parameter or a
-function that takes two. Since we don't have union types, there's no way to
-precisely type that and you'd normally have to use `dynamic`. `Function` is at
-least a *little* more precise than that.
+The new syntax is a little more verbose, but is consistent with other locations
+where you must use the new syntax.
 
 
 ### DO annotate with `Object` instead of `dynamic` to indicate any object is accepted.
 
-Some operations will work with any possible object. For example, a log method
+Some operations work with any possible object. For example, a `log()` method
 could take any object and call `toString()` on it. Two types in Dart permit all
-objects: `Object` and `dynamic`. However, they convey two different things.
+values: `Object` and `dynamic`. However, they convey different things.
 
-The `Object` annotation says "I accept any object, and I only require it to have
-the methods that `Object` itself defines."
+Using `dynamic` sends a more complex signal. It may mean that Dart's type system
+isn't sophisticated enough to represent the set of types that are allowed, or
+that the values are coming from interop or otherwise outside of the purview of
+the static type system, or you explicitly want runtime dynamism at that point in
```

Comment:
or you -> or that you
(for parallelism)

Reply:
Done.